event-driven-architecture/src/main/java/com/iluwatar/eda/framework/Channel.java

```diff
@@ -0,0 +1,11 @@
+package com.iluwatar.eda.framework;
+
+import com.iluwatar.eda.event.Event;
+
+/**
+ * Channels are delivery points for messages. Every {@link Channel} is responsible for a single type
+ * of message
+ */
+public interface Channel<E extends Message> {
+  void dispatch(Event message);
```

Comment:
According to me `dispatch` is not a appropriate name for this method. If the concrete implementations of this interface are called Handlers then shouldn't the interface itself be called `Handler`? Also the method can be renamed to `onEvent(Event e)`